[Bug] cargo install 'flutter_rust_bridge_codegen@^2.0.0-dev.2' has compilation errors

### Describe the bug

First: @fzyzcjy Gratulations on the v2 release! Will upgrade and test these days :)

When running
`cargo install 'flutter_rust_bridge_codegen@^2.0.0-dev.2'`
the following errors appear (=> I am continuing with cargo binstall):

```
   Compiling flutter_rust_bridge_codegen v2.0.0-dev.2
error[E0446]: crate-private type `CodecModePack` in public interface
  --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/func.rs:18:5
   |
18 |     pub codec_mode_pack: CodecModePack,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak crate-private type
   |
  ::: /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/generator/codec/structs.rs:13:1
   |
13 | pub(crate) struct CodecModePack {
   | ------------------------------- `CodecModePack` declared as crate-private

error[E0446]: crate-private type `lang::Lang` in public interface
  --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ident.rs:31:5
   |
31 |     pub fn style(&self, lang: &Lang) -> String {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak crate-private type
   |
  ::: /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/generator/codec/sse/lang/mod.rs:11:1
   |
11 | pub(crate) enum Lang {
   | -------------------- `lang::Lang` declared as crate-private

error[E0446]: crate-private type `lang::Lang` in public interface
  --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/namespace.rs:94:5
   |
94 |     pub fn style(&self, lang: &Lang) -> String {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak crate-private type
   |
  ::: /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/generator/codec/sse/lang/mod.rs:11:1
   |
11 | pub(crate) enum Lang {
   | -------------------- `lang::Lang` declared as crate-private

error[E0445]: crate-private trait `IrContext` in public interface
   --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/boxed.rs:12:5
    |
12  | /     fn visit_children_types<F: FnMut(&IrType) -> bool>(
13  | |         &self,
14  | |         f: &mut F,
15  | |         ir_context: &impl IrContext,
16  | |     ) {
    | |_____^ can't leak crate-private trait
    |
   ::: /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/mod.rs:151:1
    |
151 |   pub(crate) trait IrContext {
    |   -------------------------- `IrContext` declared as crate-private

error[E0445]: crate-private trait `IrContext` in public interface
   --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/dart_fn.rs:13:5
    |
13  | /     fn visit_children_types<F: FnMut(&IrType) -> bool>(
14  | |         &self,
15  | |         f: &mut F,
16  | |         ir_context: &impl IrContext,
17  | |     ) {
    | |_____^ can't leak crate-private trait
    |
   ::: /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/mod.rs:151:1
    |
151 |   pub(crate) trait IrContext {
    |   -------------------------- `IrContext` declared as crate-private

error[E0445]: crate-private trait `IrContext` in public interface
   --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/dart_opaque.rs:9:5
    |
9   | /     fn visit_children_types<F: FnMut(&IrType) -> bool>(
10  | |         &self,
11  | |         f: &mut F,
12  | |         ir_context: &impl IrContext,
13  | |     ) {
    | |_____^ can't leak crate-private trait
    |
   ::: /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/mod.rs:151:1
    |
151 |   pub(crate) trait IrContext {
    |   -------------------------- `IrContext` declared as crate-private

error[E0445]: crate-private trait `IrContext` in public interface
   --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/delegate.rs:51:5
    |
51  | /     fn visit_children_types<F: FnMut(&IrType) -> bool>(
52  | |         &self,
53  | |         f: &mut F,
54  | |         ir_context: &impl IrContext,
55  | |     ) {
    | |_____^ can't leak crate-private trait
    |
   ::: /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/mod.rs:151:1
    |
151 |   pub(crate) trait IrContext {
    |   -------------------------- `IrContext` declared as crate-private

error[E0445]: crate-private trait `IrContext` in public interface
   --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/dynamic.rs:8:5
    |
8   | /     fn visit_children_types<F: FnMut(&IrType) -> bool>(
9   | |         &self,
10  | |         _f: &mut F,
11  | |         _ir_context: &impl IrContext,
12  | |     ) {
    | |_____^ can't leak crate-private trait
    |
   ::: /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/mod.rs:151:1
    |
151 |   pub(crate) trait IrContext {
    |   -------------------------- `IrContext` declared as crate-private

error[E0445]: crate-private trait `IrContext` in public interface
   --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/enumeration.rs:48:5
    |
48  |     pub fn get<'a>(&self, file: &'a impl IrContext) -> &'a IrEnum {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak crate-private trait
    |
   ::: /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/mod.rs:151:1
    |
151 | pub(crate) trait IrContext {
    | -------------------------- `IrContext` declared as crate-private

error[E0445]: crate-private trait `IrContext` in public interface
   --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/enumeration.rs:54:5
    |
54  | /     fn visit_children_types<F: FnMut(&IrType) -> bool>(
55  | |         &self,
56  | |         f: &mut F,
57  | |         ir_context: &impl IrContext,
58  | |     ) {
    | |_____^ can't leak crate-private trait
    |
   ::: /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/mod.rs:151:1
    |
151 |   pub(crate) trait IrContext {
    |   -------------------------- `IrContext` declared as crate-private

error[E0445]: crate-private trait `IrContext` in public interface
   --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/general_list.rs:10:5
    |
10  | /     fn visit_children_types<F: FnMut(&IrType) -> bool>(
11  | |         &self,
12  | |         f: &mut F,
13  | |         ir_context: &impl IrContext,
14  | |     ) {
    | |_____^ can't leak crate-private trait
    |
   ::: /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/mod.rs:151:1
    |
151 |   pub(crate) trait IrContext {
    |   -------------------------- `IrContext` declared as crate-private

error[E0445]: crate-private trait `IrContext` in public interface
   --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/optional.rs:12:5
    |
12  | /     fn visit_children_types<F: FnMut(&IrType) -> bool>(
13  | |         &self,
14  | |         f: &mut F,
15  | |         ir_context: &impl IrContext,
16  | |     ) {
    | |_____^ can't leak crate-private trait
    |
   ::: /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/mod.rs:151:1
    |
151 |   pub(crate) trait IrContext {
    |   -------------------------- `IrContext` declared as crate-private

error[E0445]: crate-private trait `IrContext` in public interface
   --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/ownership.rs:22:5
    |
22  | /     fn visit_children_types<F: FnMut(&IrType) -> bool>(
23  | |         &self,
24  | |         f: &mut F,
25  | |         ir_context: &impl IrContext,
26  | |     ) {
    | |_____^ can't leak crate-private trait
    |
   ::: /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/mod.rs:151:1
    |
151 |   pub(crate) trait IrContext {
    |   -------------------------- `IrContext` declared as crate-private

error[E0445]: crate-private trait `IrContext` in public interface
   --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/primitive.rs:25:5
    |
25  | /     fn visit_children_types<F: FnMut(&IrType) -> bool>(
26  | |         &self,
27  | |         _f: &mut F,
28  | |         _ir_context: &impl IrContext,
29  | |     ) {
    | |_____^ can't leak crate-private trait
    |
   ::: /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/mod.rs:151:1
    |
151 |   pub(crate) trait IrContext {
    |   -------------------------- `IrContext` declared as crate-private

error[E0445]: crate-private trait `IrContext` in public interface
   --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/primitive_list.rs:11:5
    |
11  | /     fn visit_children_types<F: FnMut(&IrType) -> bool>(
12  | |         &self,
13  | |         f: &mut F,
14  | |         ir_context: &impl IrContext,
15  | |     ) {
    | |_____^ can't leak crate-private trait
    |
   ::: /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/mod.rs:151:1
    |
151 |   pub(crate) trait IrContext {
    |   -------------------------- `IrContext` declared as crate-private

error[E0445]: crate-private trait `IrContext` in public interface
   --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/record.rs:14:5
    |
14  | /     fn visit_children_types<F: FnMut(&IrType) -> bool>(
15  | |         &self,
16  | |         f: &mut F,
17  | |         ir_context: &impl IrContext,
18  | |     ) {
    | |_____^ can't leak crate-private trait
    |
   ::: /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/mod.rs:151:1
    |
151 |   pub(crate) trait IrContext {
    |   -------------------------- `IrContext` declared as crate-private

error[E0445]: crate-private trait `IrContext` in public interface
   --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/rust_auto_opaque.rs:14:5
    |
14  | /     fn visit_children_types<F: FnMut(&IrType) -> bool>(
15  | |         &self,
16  | |         f: &mut F,
17  | |         ir_context: &impl IrContext,
18  | |     ) {
    | |_____^ can't leak crate-private trait
    |
   ::: /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/mod.rs:151:1
    |
151 |   pub(crate) trait IrContext {
    |   -------------------------- `IrContext` declared as crate-private

error[E0445]: crate-private trait `IrContext` in public interface
   --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/rust_opaque.rs:28:5
    |
28  | /     fn visit_children_types<F: FnMut(&IrType) -> bool>(
29  | |         &self,
30  | |         f: &mut F,
31  | |         ir_context: &impl IrContext,
32  | |     ) {
    | |_____^ can't leak crate-private trait
    |
   ::: /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/mod.rs:151:1
    |
151 |   pub(crate) trait IrContext {
    |   -------------------------- `IrContext` declared as crate-private

error[E0445]: crate-private trait `IrContext` in public interface
   --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/structure.rs:29:5
    |
29  |     pub fn get<'a>(&self, ir_context: &'a impl IrContext) -> &'a IrStruct {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak crate-private trait
    |
   ::: /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/mod.rs:151:1
    |
151 | pub(crate) trait IrContext {
    | -------------------------- `IrContext` declared as crate-private

error[E0445]: crate-private trait `IrContext` in public interface
   --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/structure.rs:36:5
    |
36  | /     fn visit_children_types<F: FnMut(&IrType) -> bool>(
37  | |         &self,
38  | |         f: &mut F,
39  | |         _ir_context: &impl IrContext,
40  | |     ) {
    | |_____^ can't leak crate-private trait
    |
   ::: /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/mod.rs:151:1
    |
151 |   pub(crate) trait IrContext {
    |   -------------------------- `IrContext` declared as crate-private

error[E0445]: crate-private trait `IrContext` in public interface
   --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/unencodable.rs:26:5
    |
26  | /     fn visit_children_types<F: FnMut(&IrType) -> bool>(
27  | |         &self,
28  | |         _f: &mut F,
29  | |         _ir_context: &impl IrContext,
30  | |     ) {
    | |_____^ can't leak crate-private trait
    |
   ::: /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/mod.rs:151:1
    |
151 |   pub(crate) trait IrContext {
    |   -------------------------- `IrContext` declared as crate-private

error[E0445]: crate-private trait `IrContext` in public interface
   --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/mod.rs:52:5
    |
52  |     pub fn visit_types<F: FnMut(&IrType) -> bool>(&self, f: &mut F, ir_context: &impl IrContext) {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak crate-private trait
...
151 | pub(crate) trait IrContext {
    | -------------------------- `IrContext` declared as crate-private

error[E0445]: crate-private trait `IrContext` in public interface
   --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/mod.rs:80:5
    |
80  | /     fn visit_children_types<F: FnMut(&IrType) -> bool>(
81  | |         &self,
82  | |         f: &mut F,
83  | |         ir_context: &impl IrContext,
84  | |     );
    | |______^ can't leak crate-private trait
...
151 |   pub(crate) trait IrContext {
    |   -------------------------- `IrContext` declared as crate-private

error[E0445]: crate-private trait `IrContext` in public interface
   --> /Users/patmuk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-2.0.0-dev.2/src/library/codegen/ir/ty/mod.rs:78:1
    |
78  | #[enum_dispatch]
    | ^^^^^^^^^^^^^^^^ can't leak crate-private trait
...
151 | pub(crate) trait IrContext {
    | -------------------------- `IrContext` declared as crate-private
    |
    = note: this error originates in the attribute macro `enum_dispatch` (in Nightly builds, run with -Z macro-backtrace for more info)

Some errors have detailed explanations: E0445, E0446.
For more information about an error, try `rustc --explain E0445`.
error: could not compile `flutter_rust_bridge_codegen` (lib) due to 24 previous errors
warning: build failed, waiting for other jobs to finish...
error: failed to compile `flutter_rust_bridge_codegen v2.0.0-dev.2`, intermediate artifacts can be found at `/var/folders/65/4zdp9whj28j122crpwzk7d9m0000gs/T/cargo-installM9o6jl`
```

### Codegen logs with `RUST_LOG=debug` environment variable

```shell
failed before that step
```


### To Reproduce

_No response_

### Expected behavior

_No response_

### Generated binding code

_No response_

### OS

MacOS 14.2

### Version of `flutter_rust_bridge_codegen`

v2.0.0-dev.2

### Flutter info

```shell
failed before that step
```


### Version of `clang++`

_No response_

### Version of `ffigen`

_No response_

### Additional context

_No response_

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -104,6 +104,21 @@ jobs:
       - run: ./frb_internal generate-website
 
   # ----------------------------------- codegen -----------------------------------
+  msrv:
+    name: 'Test :: FRB Codegen :: MSRV'    
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install Rust
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+
+    - name: Run cargo msrv
+      run: |
+        cargo install cargo-msrv
+        cargo msrv --path frb_codegen verify
 
   generate_run_frb_codegen_command_generate:
     name: 'Generate :: FRB Codegen :: Command Generate'
@@ -146,7 +161,7 @@ jobs:
           flutter-version: ${{ env.FRB_MAIN_FLUTTER_VERSION }}
           architecture: x64
       - uses: taiki-e/install-action@cargo-llvm-cov
-
+      
       # execute
       - run: ./frb_internal generate-run-frb-codegen-command-generate --set-exit-if-changed --package ${{ matrix.package }} --coverage
 
@@ -339,7 +354,6 @@ jobs:
             version: nightly
           - image: ubuntu-latest
             version: nightly-2023-12-16
-          # TODO(@fzyzcjy, @gutenfries): add MSRV here, as well as other locked versions as needed
 
     steps:
       # setup
@@ -367,8 +381,8 @@ jobs:
       - uses: actions/upload-artifact@v4
         with:
           name: ${{ github.job }}--${{ matrix.info.image }}--${{ matrix.info.version }}--coverage
-          path: target/coverage
-
+          path: target/coverage  
+    
   test_dart_native:
     name: 'Test :: Dart :: Native'
     runs-on: ${{ matrix.image }}

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -347,8 +347,6 @@ jobs:
   test_rust:
     name: 'Test :: Rust'
     runs-on: ${{ matrix.info.image }}
-    outputs:
-        FRB_MSRV_RUST_VERSION: ${{ env.FRB_MSRV_RUST_VERSION }}
     strategy:
       fail-fast: false
       matrix:
@@ -366,9 +364,9 @@ jobs:
           - image: ubuntu-latest
             version: nightly-2023-12-16
           # tests the MSRV
-          # run on all rust packages, though only neede for frb_rust and frb_codegen
+          # run on all rust packages, though only needed for frb_rust and frb_codegen
           - image: ubuntu-latest
-          # update this, if a later MSVR is needed
+          # update this, if a later MSRV is needed
             version: 1.74.0
 
     steps:

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -365,7 +365,7 @@ jobs:
           - image: ubuntu-latest
             version: nightly-2023-12-16
           - image: ubuntu-latest
-            version: ${{ env.FRB_MSRV_RUST_VERSION }}
+            version: $env.FRB_MSRV_RUST_VERSION
 
     steps:
       # setup

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -348,6 +348,8 @@ jobs:
   test_rust:
     name: 'Test :: Rust'
     runs-on: ${{ matrix.info.image }}
+    outputs:
+        FRB_MSRV_RUST_VERSION: ${{ env.FRB_MSRV_RUST_VERSION }}
     strategy:
       fail-fast: false
       matrix:
@@ -365,7 +367,7 @@ jobs:
           - image: ubuntu-latest
             version: nightly-2023-12-16
           - image: ubuntu-latest
-            version: $env.FRB_MSRV_RUST_VERSION
+            version: ${{ needs.test_rust.outputs.FRB_MSRV_RUST_VERSION }}
 
     steps:
       # setup

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -369,7 +369,7 @@ jobs:
           # run on all rust packages, though only neede for frb_rust and frb_codegen
           - image: ubuntu-latest
           # update this, if a later MSVR is needed
-            version: 1.70.0
+            version: 1.74.0
 
     steps:
       # setup

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -10,7 +10,6 @@ on:
 env:
   CARGO_TERM_COLOR: always
   FRB_MAIN_RUST_VERSION: 1.74.0
-  FRB_MSRV_RUST_VERSION: 1.68.0
   FRB_MAIN_DART_VERSION: 3.2.0
   FRB_MAIN_FLUTTER_VERSION: 3.16.0
 
@@ -366,8 +365,11 @@ jobs:
             version: nightly
           - image: ubuntu-latest
             version: nightly-2023-12-16
+          # tests the MSRV
+          # run on all rust packages, though only neede for frb_rust and frb_codegen
           - image: ubuntu-latest
-            version: ${{ needs.test_rust.outputs.FRB_MSRV_RUST_VERSION }}
+          # update this, if a later MSVR is needed
+            version: 1.68.0
 
     steps:
       # setup

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -369,7 +369,7 @@ jobs:
           # run on all rust packages, though only neede for frb_rust and frb_codegen
           - image: ubuntu-latest
           # update this, if a later MSVR is needed
-            version: 1.68.0
+            version: 1.70.0
 
     steps:
       # setup

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -10,6 +10,7 @@ on:
 env:
   CARGO_TERM_COLOR: always
   FRB_MAIN_RUST_VERSION: 1.74.0
+  FRB_MSRV_RUST_VERSION: 1.68.0
   FRB_MAIN_DART_VERSION: 3.2.0
   FRB_MAIN_FLUTTER_VERSION: 3.16.0
 
@@ -104,21 +105,30 @@ jobs:
       - run: ./frb_internal generate-website
 
   # ----------------------------------- codegen -----------------------------------
-  msrv:
-    name: 'Test :: FRB Codegen :: MSRV'    
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v2
+  # instead of running this explicit test for frb_codegen and frb_rust only, we run the test for
+  # all rust packages in the job "test_rust". While this would not be needed for these other packages,
+  # we avoid a ci dependency to cargo-msrv.
+  # However, the code is left commented here in case this decission changes.
+  # msrv:
+  #   name: 'Test :: FRB Codegen :: MSRV'    
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #   - uses: actions/checkout@v2
 
-    - name: Install Rust
-      uses: actions-rs/toolchain@v1
-      with:
-        toolchain: stable
+  #   - name: Install Rust
+  #     uses: actions-rs/toolchain@v1
+  #     with:
+  #       toolchain: stable
 
-    - name: Run cargo msrv
-      run: |
-        cargo install cargo-msrv
-        cargo msrv --path frb_codegen verify
+  #   - name: Verify MSRV for frb_codegen
+  #     run: |
+  #       cargo install cargo-msrv
+  #       cargo msrv --path frb_codegen verify
+
+  #   - name: Verify MSRV for frb_rust
+  #     run: |
+  #       cargo install cargo-msrv
+  #       cargo msrv --path frb_rust verify
 
   generate_run_frb_codegen_command_generate:
     name: 'Generate :: FRB Codegen :: Command Generate'
@@ -354,6 +364,8 @@ jobs:
             version: nightly
           - image: ubuntu-latest
             version: nightly-2023-12-16
+          - image: ubuntu-latest
+            version: ${{ env.FRB_MSRV_RUST_VERSION }}
 
     steps:
       # setup

--- a/frb_codegen/Cargo.toml
+++ b/frb_codegen/Cargo.toml
@@ -7,7 +7,7 @@ license.workspace = true
 name = "flutter_rust_bridge_codegen"
 repository.workspace = true
 version.workspace = true
-rust-version = "1.72.0"
+rust-version = "1.74.0"
 
 [lib]
 name = "lib_flutter_rust_bridge_codegen"

--- a/frb_codegen/Cargo.toml
+++ b/frb_codegen/Cargo.toml
@@ -2,12 +2,12 @@
 categories.workspace = true
 description.workspace = true
 edition.workspace = true
+rust-version = "1.73.0"
 keywords.workspace = true
 license.workspace = true
 name = "flutter_rust_bridge_codegen"
 repository.workspace = true
 version.workspace = true
-rust-version = "1.74.0"
 
 [lib]
 name = "lib_flutter_rust_bridge_codegen"

--- a/frb_codegen/Cargo.toml
+++ b/frb_codegen/Cargo.toml
@@ -7,6 +7,7 @@ license.workspace = true
 name = "flutter_rust_bridge_codegen"
 repository.workspace = true
 version.workspace = true
+rust-version = "1.72.0"
 
 [lib]
 name = "lib_flutter_rust_bridge_codegen"

--- a/frb_codegen/Cargo.toml
+++ b/frb_codegen/Cargo.toml
@@ -2,7 +2,7 @@
 categories.workspace = true
 description.workspace = true
 edition.workspace = true
-rust-version = "1.73.0"
+rust-version = "1.74.0"
 keywords.workspace = true
 license.workspace = true
 name = "flutter_rust_bridge_codegen"

--- a/frb_rust/Cargo.toml
+++ b/frb_rust/Cargo.toml
@@ -1,7 +1,7 @@
 [package]
 name = "flutter_rust_bridge"
 edition.workspace = true
-rust-version = "1.69.0"
+rust-version = "1.70.0"
 version.workspace = true
 description.workspace = true
 license.workspace = true

--- a/frb_rust/Cargo.toml
+++ b/frb_rust/Cargo.toml
@@ -1,6 +1,7 @@
 [package]
 name = "flutter_rust_bridge"
 edition.workspace = true
+rust-version = "1.69.0"
 version.workspace = true
 description.workspace = true
 license.workspace = true

